src/main/java/executor/service/config/ThreadPoolFactory.java

```diff
@@ -0,0 +1,29 @@
+package executor.service.config;
+
+import executor.service.model.ThreadPoolConfig;
+
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+public class ThreadPoolFactory {
+    private static final int MAXIMUM_POOL_SIZE = 16;
+    private static ThreadPoolExecutor executor;
+
+    public static ThreadPoolExecutor getThreadPoolExecutor() {
+        if (executor == null) {
+            executor = configureThreadPoolExecutor();
+        }
+        return executor;
+    }
+
+    private static ThreadPoolExecutor configureThreadPoolExecutor() {
```

Comment:
Here it is better to return ExecutorService.

Reply:
If you require fine-grained control over the thread pool, such as setting specific parameters like the core pool size, maximum pool size, thread keep-alive time, then you should use ThreadPoolExecutor. This class allows you to configure the thread pool to meet your exact requirements. If you think that we dont need this, ok.